resource_work_time_from_contracts/models/resource_mixin.py

```diff
@@ -172,14 +256,28 @@ def _get_active_contracts(self, date_start, date_end):
             )
         )
 
-    def _get_work_time_per_contract(
-        self, contracts, from_datetime, to_datetime, domain
+    def _get_attendance_intervals_of_calendar(
+        self, from_datetime, to_datetime, calendar
     ):
         """
-        Return the work time per day per contract.
+        Return the attendance intervals for the provided resource.calendar,
+        ignoring leaves.
         """
-        work_time_results = []
+        return calendar._attendance_intervals(
+            from_datetime, to_datetime, self.resource_id
+        )
```

Comment:
Better to get rid of this. There's already many tiny methods, and this one seems superfluous.

Reply:
i prefer to leave it there. it’s because these kind of methods are not used in the `resource` module that we had to override so many things.